tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -34,6 +34,24 @@ sub get_script_run {
     return $pre_test_cmd;
 }
 
+sub analyzeResult {
+    my ($self, $text) = @_;
+    my $result;
+    if ($text !~ /Overall guest upgrade result is:(.*)Test done/s) {
+        die "The sub tests results are not listed in the test output. \n";
+    }
+    my $rough_result = $1;
+
+    foreach (split("\n", $rough_result)) {
+        if ($_ =~ /(?<testcase_name>\S+)\s+\-{3}\s+(PASS|FAIL|SKIP|TIMEOUT)(\s+\-{3}\s+(.*))?/ig) {
```

Comment:
Changes LGTM. The only thing I'm thinking about, is that we have multiple overrides of 'analyzeResult' and they all are 90% same, and have different statuses e.g. `PASS|FAIL|SKIP|TIMEOUT` vs `PASSED|FAILED|SKIPPED|TIMEOUT` vs `PASSED|FAILED|SKIPPED`. Is this done on purpose? I'm not saying that we have to address this in this PR, but seems that it should be possible to unify this code and have it in single place. WDYT?

Reply:
Yes, we have multiple 'analyzeResult' in our tests as you said. the test outputs are some kind of different among tests. We should consider to re-organize and standardize the log output for our the tests in our group. We will think about it in our team. thanks.